### PR TITLE
Renaming project in GUI

### LINKED ISSUE #10073: Add the ability to rename a project from inside the graph editor
Engine team believe all works on their side

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #10073 

[Screencast from 2024-06-11 13-01-10.webm](https://github.com/enso-org/enso/assets/3919101/2917ad41-e080-482b-9a69-690e80087132)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

**Comment by farmaazon** [comment]
@somebody1234 @MrFlashAccount Please, review my changes in dashboard. Also, I don't know how to properly update the name in project list after changing it in project view. Any hints?

File: app/ide-desktop/lib/dashboard/src/layouts/Editor.tsx [hunk 1]
```diff
@@ -38,6 +39,7 @@ export default function Editor(props: EditorProps) {
   const gtagEvent = gtagHooks.useGtagEvent()
   const gtagEventRef = React.useRef(gtagEvent)
   const remoteBackend = remoteBackendProvider.useRemoteBackend()
+  const { backend } = backendProvider.useStrictBackend()
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/layouts/Editor.tsx]: not sure this works - it's possible to have a Local project opened while the Cloud Backend is being used for the Drive view.

unfortunately i can't think of an easy fix... might need to add `backend` to `projectStartupInfo` (or perhaps lift the entire `renameProject` function into `projectStartupInfo` - although given that it is potentially set from multiple places it's probably unnecessarily duplicating the logic)

**Comment by somebody1234** [comment]
as for changing the project name - there is a `setProjectAsset` function in the `projectStartupInfo` - you can use it something like this

```ts
projectStartupInfo.setProjectAsset(asset => ({ ...asset, title: newName })))
```
or this, with the appropriate import (auto-import + change `{}` import to `*` import should work, or copy-paste from somewhere else with `object.merger` as a last resort)
```ts
projectStartupInfo.setProjectAsset(object.merger({ title: newName }))
```

**Comment by farmaazon** [comment]
@somebody1234 I came back to this one; applied your review with the suggestion: it works, but only when I open the project in given session; if the project is automatically opened, `setProjectAsset` is undefined.

**Comment by farmaazon** [comment]
We decided to merge this PR with the above issue. Still requires review, though.

File: app/gui2/src/components/NavBreadcrumb.vue [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
 <template>
-  <div :class="['NavBreadcrumb', { inactive: !props.active }]">
-    <span v-text="props.text"></span>
+  <div :class="['NavBreadcrumb', { inactive: !active }]">
+    <AutoSizedInput
+      v-if="editing"
+      ref="input"
+      v-model.lazy="model"
+      :autoSelect="true"
+      @change="emit('renamed', $event ?? '')"
```
Comment (kazcw) [inline app/gui2/src/components/NavBreadcrumb.vue]: When the AutoSizedInput changes, this component emits both `update:modelValue` and `renamed`?

**Review by kazcw — APPROVED** [review]

File: app/gui2/src/components/NavBreadcrumb.vue (hunk 2 above)
Comment (farmaazon) [reply app/gui2/src/components/NavBreadcrumb.vue]: Yes, the renamed event is redundant, removed.

**Review by somebody1234 — APPROVED** [review]

**Pull request merged**